lib/daru/vector.rb

```diff
@@ -603,6 +603,29 @@ def sort opts={}, &block
       Daru::Vector.new(vector, index: index, name: @name, dtype: @dtype)
     end
 
+    # Sorts the vector according to it's`Index` value. Defaults to ascending
+    # order sorting.
+    #
+    # == Options
+    #
+    # * +:ascending+ - if false, will sort `index` in descending order. Defaults to true.
+    #
+    # == Returns:
+    # new sorted `Vector` according to the index values.
+    #
+    # == Usage
+    #
+    #   dv = Daru::Vector.new [11, 12, 13], index: [23, 22, 21]
+    #   # Say you want to sort index in descending order
+    #   di.sort_by_index(ascending: false)
+    def sort_by_index opts={}
+      opts = {ascending: true, get_new_order: true}.merge(opts)
+      new_index, new_order = @index.sort(opts)
+      new_vector = new_order.map { |i| @data[i] }
```

Comment:
I think we can do this as well : 
```
new_order.each do |i|
  new_vector.push(@data[i]) 
end
```

when `new_order`  = `[2,1,0]`, `@data` = `[11, 12, 13]`
then after this loop `new_vector` = `[13,12,11]`

Is it better ?